cdap-proto/src/test/java/co/cask/cdap/proto/codec/NamespacedIdCodecTest.java

```diff
@@ -40,12 +40,12 @@
   private final Id.Namespace ns = Id.Namespace.from("ns");
   private final Id.Application app = Id.Application.from(ns, "app");
   private final Id.Program program = Id.Program.from(app, ProgramType.CUSTOM_ACTION, "action");
-  private final Id.Flow flow = Id.Flow.from(app, "flow");
```

Comment:
and less convenient. Also less type-safe. Id.Program could be abstract to remove the concern that you can have an Id.Program and an Id.Flow representing the same thing.